ecl/eclagent/eclgraph.cpp

```diff
@@ -879,10 +879,11 @@ void EclSubGraph::updateProgress()
         Owned<IWUGraphStats> progress = parent.updateStats(queryStatisticsComponentType(), queryStatisticsComponentName(), parent.queryWfid(), id);
         IStatisticGatherer & stats = progress->queryStatsBuilder();
         updateProgress(stats);
-
+        agent->mergeAggregatorStats(stats, parent.queryWfid(), parent.queryGraphName(), id);
```

Comment:
see other comments. I think with other changes, this could be something like:
```
        IStatisticGatherer & stats = progress->queryStatsBuilder();
        Owned<IStatisticCollection> sgStatsCollection = stats.getCurrentScopeCollection();
        cur.updateProgress(stats);
        agent->mergeAggregatorStats(*sgStatsCollection);
```

Reply:
That may not be true for  roxie "subgraph" stats which are for a whole graph.

Reply:
I think as per the other comments, if the "copy stats" method in the StatisticAggregator is generic and is passed an arbitrary IStatisticCollection, then uses that to discover its source scope + calculate the target to copy to, it should be able to be generic code which works in the roxie case too (with roxie similarly passing through its builder current scope level).